gcc/rust/parse/rust-parse-impl.h

```diff
@@ -4939,6 +4939,13 @@ Parser<ManagedTokenSource>::parse_trait (AST::Visibility vis,
     {
       lexer.skip_token ();
 
+      if (lexer.peek_token()->get_id() == QUESTION_MARK) {
+            lexer.skip_token();
+            rust_error_at(lexer.peek_token()->get_locus(),
+                          "%<?Trait%> is not permitted in supertraits");
+            return nullptr;
+      }
```

Comment:
yes, as @powerboat9 showed we should not be emitting this error at the parsing stage, but probably in the typechecker. you can see that it parses properly in the playground link, or with a nightly `rustc` with the `-Z parse-only` CLI option

Reply:
if you'd like any guidance for adding that check to the type checker please do say so :)

Reply:
Thanks for the feedback. I'll go ahead and try to add the check to the typechecker.